### PR TITLE
Replication

### PR DESCRIPTION
**Pull request by garrensmith**
This adds `replicate-from` and `replicate-to`. Replication to and from a
cluster to another CouchDB instance.

### CODE REVIEW

File: doc/api/nmo-replicate-from.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+nmo-replicateFrom(3) -- replicate-from
+==============================
+
+## SYNOPSIS
+
+    nmo.commands.replicateFrom([<clusterurl> || <cluster>], database, url)
```
Comment (robertkowalski) [inline doc/api/nmo-replicate-from.md]: hmm... 

we have three types of commands now:

```
nmo-couch-config     -- nmo.commands['couch-config']
nmo-activetasks      -- nmo.commands.activetasks
nmo-replicateFrom    -- nmo.commands.replicateFrom
```

(divided by dash, without dash & all small, camelCase)

we should decide for one - i would vote for all together, small (like activetasks) -- what do you think?

File: doc/cli/nmo-replicate-from.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+Example:
+
+This will replicate the database `hello-database` from the cluster `mycluster` to the supplied url
+    nmo replicate-from mycluster hello-database http://192.0.0.1/hello-database-replicated
+
+This will replicate the database `hello-database` from the cluster `mycluster` to the supplied url and create the target and make it a continuous replication
+    nmo replicate-from mycluster hello-database http://192.0.0.1/hello-database-replicated --create-target --continuous
+
+This will replicate the database `hello-database` from the url `http://my-couchdb-cluster.com` to the supplied url
+    nmo replicate-from http://my-couchdb-cluster hello-database http://192.0.0.1/hello-database-replicated
```
Comment (robertkowalski) [inline doc/cli/nmo-replicate-from.md]: url must be `http://my-couchdb-cluster.com`

File: doc/cli/replicate-to.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+Example:
+
+This will replicate the database `hello-database` from the url to database `hlloe-database-replicated` in cluster `mycluster`.
+    nmo replicate-to mycluster hello-database-replicated http://192.0.0.1/hello-database
+
+This will replicate the database `hello-database` to the cluster `mycluster` and create the database and make it a continuous replication
+    nmo replicate-to mycluster hello-database-replicated http://192.0.0.1/hello-database --create-target --continuous
+
+This will replicate the database `hello-database` to the url `http://my-couchdb-cluster.com` from the supplied url
+    nmo replicate-from http://my-couchdb-cluster hello-database-replicated http://192.0.0.1/hello-database
```
Comment (robertkowalski) [inline doc/cli/replicate-to.md]: replicate-to?

File: doc/api/nmo-replicate-from.md (hunk 1 above)
Comment (robertkowalski) [inline doc/api/nmo-replicate-from.md]:
```
nmo.commands.replicateFrom([<clusterurl> || <cluster>], database, url)
```

can the last argument (url) also be a cluster name?

Comment (garrensmith) [inline doc/api/nmo-replicate-from.md]: No the last one has to be a url for now.

File: doc/api/nmo-replicate-from.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+nmo-replicateFrom(3) -- replicate-from
```
Comment (robertkowalski) [inline doc/api/nmo-replicate-from.md]: replicatefrom

File: doc/api/replicate-to.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+nmo-replicateTo(3) -- replicate-to
```
Comment (robertkowalski) [inline doc/api/replicate-to.md]: replicateto

File: src/replicate-to.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+export function cli (cluster, dbname, url) {
+  return new Promise((resolve, reject) => {
+    if (!cluster) {
+      const msg = [
+        'Usage:',
+        '',
+        'nmo replicate-to <cluster> <databasename> <url/dbname> [--continuous] [--create-target] [--json]',
+        'nmo replicate-to <url> <databasename> <url/dbname> [--continuous] [--create-target] [--json]',
+      ].join('\n');
```
Comment (robertkowalski) [inline src/replicate-to.js]: we should test for the dbname and url too, example of my wrong usage:

```
(13:04:02) [robert@tequila-work] ~/apache/couchdb-nmo (pr/11) $ ./bin/nmo-cli.js replicate-to anemone https://examples.cloudant.com/animaldb
http request POST http://127.0.0.1:15984/_replicator
http 403 http://127.0.0.1:15984/_replicator
```

Comment (garrensmith) [inline src/replicate-to.js]: Yes good point.

File: src/replicate-from.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
+import nmo from './nmo.js';
+import Promise from 'bluebird';
+import { getUrlFromCluster } from './utils';
+import {replicate, createReplicatorDoc } from './replicate';
+
+
+export function cli (cluster, dbname, url) {
+  return new Promise((resolve, reject) => {
```
Comment (robertkowalski) [inline src/replicate-from.js]: we should test for the dbname and url too (see other command for example)

**Comment by robertkowalski** [comment]
looks good, merge after fixing the small nits i found :)

**Comment by garrensmith** [comment]
Thanks for reviewing @robertkowalski. Merged.

**Pull request closed without merging**